DiffMatchPatch.zig

```diff
@@ -587,28 +726,31 @@ fn diffLineMode(
         switch (diffs.items[pointer].operation) {
             .insert => {
                 count_insert += 1;
-                // text_insert += diffs.items[pointer].text;
                 try text_insert.appendSlice(allocator, diffs.items[pointer].text);
             },
             .delete => {
                 count_delete += 1;
-                // text_delete += diffs.items[pointer].text;
                 try text_delete.appendSlice(allocator, diffs.items[pointer].text);
             },
             .equal => {
                 // Upon reaching an equality, check for prior redundancies.
                 if (count_delete >= 1 and count_insert >= 1) {
                     // Delete the offending records and add the merged ones.
-                    // diffs.RemoveRange(pointer - count_delete - count_insert, count_delete + count_insert);
+                    freeRangeDiffList(
+                        allocator,
+                        &diffs,
+                        pointer - count_delete - count_insert,
+                        count_delete + count_insert,
+                    );
                     try diffs.replaceRange(
```

Comment:
replacing a range with zero elements will never fail. This should call `replaceRangeAssumeCapacity` instead. If this function would actually return an error then a double free would likely occur.

There are few other `replaceRange` calls that would also be eligible for this change.

Reply:
Done but like. You'd think that ArrayLists would have a better way to spell `deleteRange` than `replaceRangeAssumeCapacity`...

Reply:
There's 26 letters in there. Even with the gutter at 100 that's a quarter of a line. Anyway. Separate issue.